pkg/clients/openshift.go

```diff
@@ -45,29 +49,27 @@ func (t *TokenReview) Groups() []string {
 
 //Namespace wrappers a core kube namespace type
 type Namespace struct {
-	corev1.Namespace
+	Ns corev1.Namespace
 }
 
 //UID get the UID of a namespace
 func (ns *Namespace) UID() string {
-	return ns.UID()
+	return string(ns.Ns.UID)
 }
 
 //Name get the name of a namespace
 func (ns *Namespace) Name() string {
-	return ns.Name()
+	return ns.Ns.Name
 }
 
 //ListNamespaces associated with a given token
 func (c *DefaultOpenShiftClient) ListNamespaces(token string) (namespaces []Namespace, err error) {
-
-	var nsList *corev1.NamespaceList
-	if nsList, err = newKubeClient(token).CoreV1().Namespaces().List(metav1.ListOptions{}); err != nil {
```

Comment:
@jcantrill you had this wrong.

I have fixed this to return `nil, err` in case of an error and return a list of ns is call passes. However this call passes only if the user as cluster-admin role: `oc adm policy add-cluster-role-to-user cluster-admin developer5`.

```
{"code":500,"message":"Internal Error","error":{"ErrStatus":{"metadata":{},"status":"Failure","message":"namespaces is forbidden: User \"developer5\" cannot list namespaces at the cluster scope: no RBAC policy matched","reason":"Forbidden","details":{"kind":"namespaces"},"code":403}}}
```

Reply:
I haven't find a way how to return a list of namespaces belonging to a user without requiring `cluster-admin` role. 

Is it expected that user will have cluster admin role? Maybe we could use the ES service account to get all namespaces and then filter out for a specific user.

Reply:
If we have a user's token then we can get their list of projects using their token.  Is it possible that fetching namespaces is different from fetching projects?  It's crucial we resolve this otherwise we will be unable to resolve seeding.  If something you are unable to resolve then we should open a GH issue to investigate as project fetching is not relevant to the cache code.

Reply:
I was expected that you deployed this proxy and tested before so it eventually worked. Maybe just my local configuration was not working with this.

I can have a look at it in a separate ticket - this is not related to cache at all.

Reply:
https://github.com/openshift/elasticsearch-proxy/issues/5